lxd/db/update.go

```diff
@@ -91,14 +97,72 @@ func UpdatesApplyAll(db *sql.DB, doBackup bool, postApply func(int) error) error
 			backup = true
 		}
 		logger.Debugf("Updating DB schema from %d to %d", version, version+1)
-		if postApply != nil {
-			return postApply(version + 1)
+		if hook != nil {
+			// XXX We need to commit the transaction before the
```

Comment:
YMMV, in Landscape we used to put "XXX" at the beginning of a comment to mean "the following sentence describes something unfortunate, that I wish I could avoid or do differently". Not sure if there's any convention like that in LXD (or if perhaps "XXX" is used differently).